addons/account/wizard/account_payment_register.py

```diff
@@ -716,9 +721,8 @@ def _fetch_duplicate_reference(self, matching_states=('draft', 'posted')):
                )
         """,
             move_table_and_alias=move_table_and_alias,
-            matching_states=tuple(matching_states),
-            suspense_account_id=self.company_id.account_journal_suspense_account_id.id,
-            outstanding_account_ids=outstanding_account_ids,
+            matching_states=list(matching_states),
+            additional_account_ids=self.company_id.account_journal_suspense_account_id.ids + outstanding_account_ids,
```

Comment:
we should be checking the journal's suspense account id (which will be probably be the company's default